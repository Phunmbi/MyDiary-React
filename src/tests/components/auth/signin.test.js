import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import SigninComponent, { Signin } from '../../../components/auth/Signin';
import Root from '../../../root';

let wrapper;
let wrapped;
let wrappedTrue;

const response = {
  auth: {
    error: false,
  },
  addEntry: jest.fn(),
  entries: {
    newEntryError: {
      data: {
        message: 'new',
      },
      length: 5,
    },
  },
};

const history = {
  push: jest.fn(),
};

const responseTrue = {
  auth: {
    status: 200,
    error: {
      data: {
        Error: true,
      },
    },
  },
  addEntry: jest.fn(),
  entries: {
    newEntryError: {
      data: {
        message: 'new',
      },
      length: 5,
    },
  },
};

document.getElementById = jest.fn((selector) => {
  console.log('selector => ', selector);

  return {
    style: {
      background: 'background',
    },
    innerHTML: 'inner',
    disabled: true,
  };
});

beforeEach(() => {
  jest.useFakeTimers();
  wrapper = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
        <SigninComponent
          response={response}
        />
      </MemoryRouter>
    </Root>,
  );

  wrapped = shallow(
    <Signin
      response={response}
    />,
  );


  wrappedTrue = shallow(
    <Signin
      response={responseTrue}
      history={history}
    />,
  );
});

afterEach(() => wrapper.unmount());

describe('Signin UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Signin Functionality', () => {
  it('should click simulate submit', () => {
    const submitButton = wrapped.find('form');
    const email = wrapped.find('#email');

    const mockedEvent = {
      preventDefault: jest.fn(),
      target: {
        reset: jest.fn(),
        children: {
          firstName: {
            value: 'value',
          },
          lastName: {
            value: 'value',
          },
          email: {
            value: 'value',
          },
          password: {
            value: 'value',
          },
          confirmPassword: {
            value: 'value',
          },
          title: 'title',
          details: 'details',
          submit: {
            innerHTML: 'submit',
            style: {
              background: 'background',
            },
          },
        },
      },
    };

    submitButton.simulate('submit', mockedEvent);
    email.simulate('change', mockedEvent);
  });

  it('should click simulate submit', () => {
    const submitButton = wrappedTrue.find('form');
    const email = wrappedTrue.find('#email');

    const mockedEvent = {
      preventDefault: jest.fn(),
      target: {
        reset: jest.fn(),
        children: {
          firstName: {
            value: 'value',
          },
          lastName: {
            value: 'value',
          },
          email: {
            value: 'value',
          },
          password: {
            value: 'value',
          },
          confirmPassword: {
            value: 'value',
          },
          title: 'title',
          details: 'details',
          submit: {
            innerHTML: 'submit',
            style: {
              background: 'background',
            },
          },
        },
      },
    };

    submitButton.simulate('submit', mockedEvent);
    email.simulate('change', mockedEvent);
  });
});
