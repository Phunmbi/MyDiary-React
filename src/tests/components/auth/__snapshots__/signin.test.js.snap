// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin UI render features component should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testkey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testkey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testkey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Signin)
          response={
            Object {
              "addEntry": [MockFunction],
              "auth": Object {
                "error": false,
              },
              "entries": Object {
                "newEntryError": Object {
                  "data": Object {
                    "message": "new",
                  },
                  "length": 5,
                },
              },
            }
          }
        >
          <Signin
            response={
              Object {
                "auth": Object {
                  "authenticated": null,
                  "firstName": null,
                  "lastName": null,
                },
                "entries": Object {
                  "authenticated": "",
                  "deletedEntry": Array [],
                  "deletedEntryError": Array [],
                  "deletedReminder": Array [],
                  "deletedReminderError": Array [],
                  "editedEntry": Array [],
                  "editedEntryError": Array [],
                  "entries": Array [],
                  "entriesError": Array [],
                  "error": Array [],
                  "existingReminder": Array [],
                  "existingReminderError": Array [],
                  "firstName": "",
                  "lastName": "",
                  "newEntry": Array [],
                  "newEntryError": Array [],
                  "newReminder": Array [],
                  "newReminderError": Array [],
                  "singleEntry": Array [],
                  "singleEntryError": Array [],
                  "status": "",
                },
                "profile": Object {
                  "authenticated": "",
                  "deletedEntry": Array [],
                  "deletedEntryError": Array [],
                  "deletedReminder": Array [],
                  "deletedReminderError": Array [],
                  "editedEntry": Array [],
                  "editedEntryError": Array [],
                  "entries": Array [],
                  "entriesError": Array [],
                  "error": Array [],
                  "existingReminder": Array [],
                  "existingReminderError": Array [],
                  "firstName": "",
                  "lastName": "",
                  "newEntry": Array [],
                  "newEntryError": Array [],
                  "newReminder": Array [],
                  "newReminderError": Array [],
                  "singleEntry": Array [],
                  "singleEntryError": Array [],
                  "status": "",
                },
              }
            }
            signin={[Function]}
          >
            <div
              className="hero"
            >
              <div
                className="title"
              >
                <h1>
                  MyDiary
                </h1>
              </div>
              <section>
                <div
                  className="hero-marketing"
                >
                  <p>
                    Let's help you make it easier to keep your diary. 
                  </p>
                  <p>
                    With daily programmable reminders. 
                  </p>
                  <p>
                    The ability to edit and delete entries as you please.
                  </p>
                  <p>
                    All wrapped in a lightweight app built for your convenience.
                  </p>
                </div>
                <div
                  className="hero-account"
                >
                  <h3>
                    Sign in
                  </h3>
                  <p
                    className="errorResponse"
                    id="errorResponse"
                  />
                  <form
                    onSubmit={[Function]}
                  >
                    <input
                      id="email"
                      onChange={[Function]}
                      placeholder="Email"
                      required={true}
                      type="email"
                    />
                    <span
                      className="errorResponse"
                    />
                    <input
                      id="password"
                      min="8"
                      onChange={[Function]}
                      placeholder="Password"
                      required={true}
                      type="password"
                    />
                    <span
                      className="errorResponse"
                    />
                    <button
                      id="submit"
                      type="submit"
                    >
                      Sign in
                    </button>
                  </form>
                  <p>
                    Don't have an account? then
                    <Link
                      replace={false}
                      to="/"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                         Sign up
                      </a>
                    </Link>
                  </p>
                </div>
              </section>
            </div>
          </Signin>
        </Connect(Signin)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
